### PR TITLE
Added note in docs for change listeners

### LINKED ISSUE #927: addListener notification is empty (documentation only)
When using addListener(objects, changes) on React Native iOS it always returns empty on the first notification and the second notification returns what should have been in the first one.

I found in results-tests.js below that first notification is empty. 

Is this by design or am I doing something wrong here?  :grinning:

```
        realm.objects('TestObject').addListener((testObjects, changes) => {
            // TODO: First notification is empty, so perform these
            // assertions on the second call. However, there is a race condition
            // in React Native, so find a way to do this in a robust way.
            //TestCase.assertEqual(testObjects.length, 4);
            //TestCase.assertEqual(changes.insertions.length, 1);
            resolve();
        });
```

### PR DESCRIPTION
**Pull request by papafe**
This closes #927

### CODE REVIEW

File: packages/realm/src/Object.ts [hunk 1]
```diff
@@ -404,6 +404,7 @@ export class RealmObject<T = DefaultObject> {
    *      console.log(` ${prop}`);
    *   });
    * })
+   * @note The callback will be invoked with an empty `changes` when the listener is registered.
```
Comment (elle-j) [inline packages/realm/src/Object.ts:407]:
```suggestion
   * @note The callback will also be invoked when the listener is added, containing an empty array for `changes.changedProperties`.
```

File: packages/realm/src/Collection.ts [hunk 2]
```diff
@@ -124,6 +124,7 @@ export abstract class Collection<
    *  console.log(`${changes.deletions.length} deletions`);
    *  console.log(`new size of collection: ${collection.length}`);
    * });
+   * @note The callback will be invoked with an empty `changes` when the listener is registered.
```
[the remark this answers is not in the document]
Comment (papafe) [reply packages/realm/src/Collection.ts:127]: You are definitely correct :)

**Review by kraenhansen — APPROVED** [review]
I believe this applies for the `Realm#addListener` too.
Perhaps it's worth mentioning why this is .. specifically that adding a listener is an asynchronous operation and the first event is fired for the caller to know when the listener has been added. (at least that was the explanation @tgoyne once gave to me).

**Comment by papafe** [comment]
> I believe this applies for the `Realm#addListener` too. Perhaps it's worth mentioning why this is .. specifically that adding a listener is an asynchronous operation and the first event is fired for the caller to know when the listener has been added. (at least that was the explanation @tgoyne once gave to me).

I agree with adding the explanation. I've tried running some tests using `Realm.addListener`, and it seems it doesn't apply to that.

**Pull request merged**